### PR TITLE
Remove irrelevant targets from cache

### PR DESCRIPTION
**Pull request by garethbowen**
#3207

### CODE REVIEW

File: static/js/services/target-generator.js [hunk 1]
```diff
@@ -54,10 +54,12 @@ var moment = require('moment'),
           // unconfigured target type
           return;
         }
-        if (instance.deleted) {
-          delete target.instances[instance._id];
-        } else if (isRelevant(instance)) {
+        if (isRelevant(instance) && !instance.deleted) {
```
Comment (alxndrsn) [inline static/js/services/target-generator.js]: Probably cheaper to check `.deleted` before calling `isRelevant()`, but anyway if an instance is deleted, it's presumably not relevant.  So I'd expect `isRelevant()` to perform the deleted check.

File: tests/karma/unit/services/target-generator.js [hunk 2]
```diff
@@ -338,4 +338,41 @@ describe('TargetGenerator service', function() {
...
+    UserContact.returns(KarmaUtils.mockPromise());
+    var callbackCount = 0;
+    injector.get('TargetGenerator')(function(err, actual) {
+      if (callbackCount === 0) {
+        chai.expect(actual.length).to.equal(1);
+        chai.expect(actual[0].id).to.equal('report');
+        chai.expect(actual[0].type).to.equal('count');
+        chai.expect(actual[0].count).to.equal(1);
+      } else if (callbackCount === 1) {
+        chai.expect(actual.length).to.equal(1);
```
Comment (alxndrsn) [inline tests/karma/unit/services/target-generator.js]: I don't get why it still generates a target here, but the target has no count.

File: static/js/services/target-generator.js (hunk 1 above)
Comment (garethbowen) [inline static/js/services/target-generator.js]: Good call. I'll make that change.

File: tests/karma/unit/services/target-generator.js (hunk 2 above)
Comment (garethbowen) [inline tests/karma/unit/services/target-generator.js]: Targets are your monthly achievements such as "register 10 pregnancies". The count is how much progress you've made towards your target. These are shown on the Targets page even if you have made no progress towards your achievement yet. So in this case the target would show on the page with 0 pregnancies registered.

**Pull request merged**